app/scripts/modules/core/src/entityTag/EntityTagEditor.tsx

```diff
@@ -0,0 +1,322 @@
+      onTaskComplete: () => this.props.onUpdate(),
+    });
+
+    const submitMethod = () => {
+      const promise = this.entityTagWriter.upsertEntityTag(application, tag, entityRef, isNew);
+      const done = () => this.setState({ isSubmitting: false });
+      promise.then(done, done);
+      return promise;
+    };
+
+    taskMonitor.submit(submitMethod);
+
+    this.setState({ taskMonitor, isSubmitting: true });
+  }
+
+  public render(): React.ReactElement<any> {
+    const { isNew, tag, ownerOptions } = this.props;
+    const message = this.state.message || '';
+
+    const closeButton = (
+      <div className="modal-close close-button pull-right">
+        <a className="btn btn-link" onClick={this.onHide}>
+          <span className="glyphicon glyphicon-remove" />
+        </a>
+      </div>
+    );
+
+    const { TaskMonitorWrapper } = NgReact;
+
+    return (
+      <Modal show={this.state.show} onHide={this.onHide} dialogClassName="entity-tag-editor-modal">
+
+        <TaskMonitorWrapper monitor={this.state.taskMonitor} />
+
+        <Formsy.Form
+          role="form"
+          name="form"
+          className="form-horizontal"
+          onSubmit={this.upsertTag}
+          onValid={this.onValid}
+          onInvalid={this.onInvalid}
+        >
+          <Modal.Header>
+            <h3>{isNew ? 'Create' : 'Update'} {tag.value.type}</h3>
+            {closeButton}
+          </Modal.Header>
+
+          <Modal.Body>
+            <EntityTagMessage
+              message={message}
+              onMessageChanged={this.handleMessageChanged}
+            />
+
+            { ownerOptions && ownerOptions.length && (
+              <OwnerOptions
+                selectedOwner={this.state.owner}
+                ownerOptions={ownerOptions}
+                onOwnerOptionChanged={this.handleOwnerOptionChanged}
+              />
+            ) }
+          </Modal.Body>
+
+          <Modal.Footer>
+            <button type="button" className="btn btn-default" onClick={this.onHide}>Cancel</button>
+            <button type="submit" className="btn btn-primary" disabled={!this.state.isValid || this.state.isSubmitting}>
+              <span className="glyphicon glyphicon-ok-circle" />
+              <span>{this.props.isNew ? ' Create' : ' Update'} {this.props.tag.value.type}</span>
+            </button>
+          </Modal.Footer>
+        </Formsy.Form>
+
+      </Modal>
+    );
+  }
+}
+
+
+
+interface IEntityTagMessageProps {
+  message: string;
+  onMessageChanged(message: string): void;
+}
+
+@autoBindMethods
+class EntityTagMessage extends React.Component<IEntityTagMessageProps, {}> {
+  private handleTextareaChanged(event: React.FormEvent<any>): void {
+    this.props.onMessageChanged(event.currentTarget.value);
+  }
+
+  public render(): React.ReactElement<any> {
+    const { message } = this.props;
+
+    return (
+      <div className="row">
+        <div className="col-md-10 col-md-offset-1">
+
+          <TextArea
```

Comment:
Each validation component type has to be written by us.  

The `label` is passed as a string and wrapped in a `<label>` tag by the TextArea component.  In contrast, `_help` is passed as a react element, and includes markup.